src/main/java/com/connexta/store/poller/TransformStatusPoller.java

```diff
@@ -0,0 +1,281 @@
+              transformPollUrl);
+          return true;
+        }
+      } else if (HttpStatus.NOT_FOUND.equals(httpStatus)) {
+        log.debug(
+            "Transform URL status for {} does not exist anymore, removing from task queue.",
+            transformPollUrl);
+        return false;
+      } else {
+        // for the rest of responses, we will just retry
+        return true;
+      }
+    } else {
+      log.debug(
+          "Did not receive a response from transformation status {}. This URL will be retried.",
+          transformPollUrl);
+      return true;
+    }
+  }
+
+  private boolean handlePollResponse(
+      String datasetId, TransformationPollResponse transformationPollResponse) {
+    final Status transformationStatus = transformationPollResponse.getTransformationStatus();
+    switch (transformationStatus) {
+      case DONE:
+        return addMetadata(datasetId, transformationPollResponse.getMetadataInformations());
+      case FAILED:
+        logFailedTransformation(datasetId, transformationPollResponse.getMetadataInformations());
+        return quarantine(datasetId, transformationPollResponse.getMetadataInformations());
+      case IN_PROGRESS:
+      default:
+        log.debug("Transformation in progress for Dataset {}", datasetId);
+        return true;
+    }
+  }
+
+  private void logFailedTransformation(
+      String datasetId, List<MetadataInformation> metadataInformations) {
+    log.debug("Metadata transformation failed for dataset {}", datasetId);
+    for (MetadataInformation metadataInformation : metadataInformations) {
+      log.debug("Metadata: {}", metadataInformation);
+    }
+  }
+
+  /** @return {@code true} if the add metadata should be retried, {@code false} otherwise */
+  private boolean addMetadata(String datasetId, List<MetadataInformation> metadataInformations) {
+    AddMetadataRequest addMetadataRequest = new AddMetadataRequest();
+    for (MetadataInformation metadataInformation : metadataInformations) {
+      addMetadataRequest.addMetadataInfo(
+          new MetadataInfo()
+              .metadataType(metadataInformation.getMetadataType())
+              .location(metadataInformation.getLocation()));
+    }
+
+    ClientResponse response =
+        storeWebClient
+            .put()
+            .uri(
+                UriComponentsBuilder.fromPath(StoreController.ADD_METADATA_URL_TEMPLATE)
+                    .build(datasetId)
+                    .toASCIIString())
+            .body(BodyInserters.fromValue(addMetadataRequest))
+            .exchange()
+            .block();
+
+    if (response != null) {
+      final HttpStatus httpStatus = response.statusCode();
+      if (HttpStatus.OK.equals(httpStatus)) {
+        deleteTransformationStatus(datasetId);
+        return false;
+      } else if (HttpStatus.NOT_FOUND.equals(httpStatus)) {
+        log.debug("Failed to add metadata to dataset {} because it no longer exists.", datasetId);
```

Comment:
✏️ This seems like another of those catastrophic cases where a stronger log may be helpful